### PR TITLE
Feature/molecule badge counter

### PR DESCRIPTION
**Pull request by juanmaguitar**
- ⚙️ Task → https://jira.scmspain.com/browse/SUIC-23
- 📄 Paper → https://paper.dropbox.com/doc/SUI-Badge-Counter-WIP--AON_E_s4jkzUsufM4AldI8sdAg-Ng50gIyG0zIXbZ6IuHcsq
- 🌍 Demo Online → https://sui-components-feature-molecule-badge-counter.now.sh/workbench/molecule/badgeCounter/

✅ OK por parte de UX → https://jira.scmspain.com/browse/SUIC-23?focusedCommentId=799595&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-799595

### CODE REVIEW

**Review by jcger — CHANGES_REQUESTED** [review]

File: components/molecule/badgeCounter/src/index.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+
+const BASE_CLASS = `sui-MoleculeBadgeCounter`
+const CLASS_WITH_CHILDREN = `${BASE_CLASS}--with-children`
```
[the remark this answers is not in the document]
Comment (midudev) [reply components/molecule/badgeCounter/src/index.js]: `${BASE_CLASS}--withChildren`

File: components/molecule/badgeCounter/src/index.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    <span className={className}>
+      <span className={classNameBullet}>{processedLabel}</span>
+      <span>{children}</span>
+    </span>
+  )
+}
+
+MoleculeBadgeCounter.displayName = 'MoleculeBadgeCounter'
+
+// Remove these comments if you need
```
Comment (midudev) [inline components/molecule/badgeCounter/src/index.js]: 👀

File: components/molecule/badgeCounter/src/index.scss [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    background-color: $bg-molecule-badge-counter;
+    border-radius: 100%;
+    color: $c-molecule-badge-counter;
+    display: inline-flex;
+    font-size: $fz-xxs;
+    justify-content: center;
+
+    
+    &-small {
+      height: 6px;
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.scss]: done!

File: components/molecule/badgeCounter/src/index.scss [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+      .sui-MoleculeBadgeCounter--with-children & {
+        position: absolute;
+        right: -5px;
+        top: 0;
+      }
+  
+      &-dot {
+        background-color: $c-white;
+        border: 2px solid red;
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.scss]: done!

File: components/molecule/badgeCounter/src/index.scss [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    }
+  
+    &-medium {
+      height: 12px;
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.scss]: done!

File: components/molecule/badgeCounter/src/index.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  /** children */
+  children: PropTypes.any,
+
+  /** label */
+  label: PropTypes.any,
+
+  /** size */
+  size: PropTypes.any,
+
+  /** variant */
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.js]: done!

File: components/molecule/badgeCounter/src/index.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+MoleculeBadgeCounter.displayName = 'MoleculeBadgeCounter'
+
+// Remove these comments if you need
+// MoleculeBadgeCounter.contextTypes = {i18n: PropTypes.object}
+MoleculeBadgeCounter.propTypes = {
+  /** children */
+  children: PropTypes.any,
+
+  /** label */
+  label: PropTypes.any,
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.js]: done!

File: components/molecule/badgeCounter/src/index.scss [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+      &-exclamation {
+        &::after {
+          content: '!';
+        }
+      }
+
+      &-two-characters,
+      &-three-characters {
+        padding: 0 2px;
+        border-radius: 15px;
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.scss]: done!

File: components/molecule/badgeCounter/src/index.scss [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+      &-exclamation {
+        &::after {
+          content: '!';
+        }
+      }
+
+      &-two-characters,
+      &-three-characters {
+        padding: 0 2px;
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.scss]: done!

File: components/molecule/badgeCounter/src/index.scss [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+      .sui-MoleculeBadgeCounter--with-children & {
+        position: absolute;
+        right: -12px;
+        top: 0;
+      }
+  
+      &-dot {
+        background-color: $c-white;
+        border: 4px solid red;
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.scss]: done!

File: components/molecule/badgeCounter/src/index.js [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+
+const BASE_CLASS = `sui-MoleculeBadgeCounter`
+const CLASS_WITH_CHILDREN = `${BASE_CLASS}--with-children`
+
+const CLASS_BULLET = `${BASE_CLASS}-bullet`
+const CLASS_SMALL = `${CLASS_BULLET}-small`
+const CLASS_MEDIUM = `${CLASS_BULLET}-medium`
+const CLASS_MEDIUM_ONE_CHAR = `${CLASS_MEDIUM}-one-character`
```
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.js]: done!

File: components/molecule/badgeCounter/src/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.js]: done!

File: components/molecule/badgeCounter/src/index.js (hunk 2 above)
Comment (juanmaguitar) [reply components/molecule/badgeCounter/src/index.js]: done!

File: components/molecule/badgeCounter/README.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  size={moleculeBadgeCounterSizes.MEDIUM}
+/>
+```
+
+### Labels
+
+```js
+<MoleculeBadgeCounter label="4" />
+<MoleculeBadgeCounter label="42" />
+<MoleculeBadgeCounter label="+42" />
```
Comment (jcger) [inline components/molecule/badgeCounter/README.md]: this wouldn't work as a number is required now, right?

**Review by jcger — DISMISSED** [review]

File: components/molecule/badgeCounter/README.md (hunk 12 above)
Comment (juanmaguitar) [reply components/molecule/badgeCounter/README.md]: Yes, that wouldn't work. I updated the demo but I didn't update the README. It's updated now.

**Review by jcger — APPROVED** [review]

File: components/molecule/badgeCounter/src/index.js [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+    size === SIZES.MEDIUM || hasLabel ? CLASS_MEDIUM : CLASS_SMALL
+  const CLASS_LENGTH_LABEL =
+    hasLabel && (label + '').length >= 3 ? CLASS_MEDIUM_THREE_CHARS : ''
+  const CLASS_VARIANT = variant ? `${CLASS_SIZE}-${variant}` : ''
+
+  const className = cx(BASE_CLASS, {
+    [CLASS_WITH_CHILDREN]: hasChildren
+  })
+
+  const processedLabel = label.length >= 3 ? '+99' : label
```
Comment (kikoruiz) [inline components/molecule/badgeCounter/src/index.js:38]: Maybe we could move these "magic numbers" into constants.

**Review by kikoruiz — APPROVED** [review]

**Review by adriadoz — APPROVED** [review]

**Pull request merged**